src/stores/SetupEncryptionStore.js

```diff
@@ -85,7 +87,22 @@ export class SetupEncryptionStore extends EventEmitter {
             this.keyInfo = keys[this.keyId];
         }
 
-        this.phase = PHASE_INTRO;
+        // do we have any other devices which are E2EE which we can verify against?
+        const dehydratedDevice = await cli.getDehydratedDevice();
+        this.hasDevicesToVerifyAgainst = cli.getStoredDevicesForUser(cli.getUserId()).some(
+            device =>
+                device.getIdentityKey() &&
+                (!dehydratedDevice || (device.deviceId != dehydratedDevice.device_id))
+        );
+
+        if (!this.hasDevicesToVerifyAgainst && !this.keyInfo) {
+            // skip before we can even render anything.
+            // XXX: this causes a dialog box flash
```

Comment:
As in the verification prompt appears briefly before we realise there are no devices...? Perhaps we can use a "loading" phase first, then we go to either intro or finished based on the results?

Reply:
done.